core/bcast/bcast.go

```diff
@@ -125,6 +125,66 @@ func (b Broadcaster) Broadcast(ctx context.Context, duty core.Duty,
 	}
 }
 
+// Broadcast2 broadcasts the aggregated signed duty data object to the beacon-node.
+func (b Broadcaster) Broadcast2(ctx context.Context, duty core.Duty,
+	pubkey core.PubKey, aggData core.SignedData,
+) (err error) {
+	ctx = log.WithTopic(ctx, "bcast")
+	ctx = log.WithCtx(ctx, z.Any("pubkey", pubkey))
+	defer func() {
+		if err == nil {
+			instrumentDuty(duty, pubkey, b.delayFunc(duty.Slot))
+		}
+	}()
+
+	v := signedDataVisitor{eth2Cl: b.eth2Cl, delayFunc: b.delayFunc}
+
+	return aggData.VisitDuty(ctx, duty, v)
+}
+
+type signedDataVisitor struct {
```

Comment:
the nice thing is the each type's logic is defined in its own function. The pad thing is that we cannot define other input parameters our output parameters to the function since Go generics doesn't support "function generics", it only support "type generics"...